### PR TITLE
Pullquote: Fix parsing/serializing multi-paragraphs in pullquotes

### LINKED ISSUE #1982: The pullquote block outputs nested paragraphs
When inserting a Pullquote block the produced output contains nested paragraphs.
Just insert a Pull quote and the initial markup will be something like:

```
<!-- wp:core/pullquote -->
<blockquote class="wp-block-pullquote alignnone">
    <p>
        <p>Pull quote content</p>
    </p>
    <footer>Pull quote caption</footer>
</blockquote>
<!-- /wp:core/pullquote -->
```

Save, and navigate away, then re-open the post in Gutenberg (or just refresh the page). TinyMCE will try to fix the markup. This will end up in something like:
```
<!-- wp:core/pullquote -->
<blockquote class="wp-block-pullquote alignnone">
    <p>
    </p>
    <p>Pull quote content</p>
    <p></p>
    <footer>Pull quote caption</footer>
</blockquote>
<!-- /wp:core/pullquote -->
```

Additionally: seems the empty paragraphs above are preserved in Gutenberg. However, when I try this:
- navigate away and re-open the post in the classic editor - visual mode
- save or switch to text mode then back to visual
- the markup will change in something like this:
```
<!-- wp:core/pullquote -->
<blockquote class="wp-block-pullquote alignnone">Pull quote content

<footer>Pull quote caption</footer></blockquote>
<!-- /wp:core/pullquote -->
```

- save and navigate away
- re-open the post in Gutenberg
- now the pullquote content has disappeared, the markup will be something like this:
```
<!-- wp:core/pullquote -->
<blockquote class="wp-block-pullquote alignnone">
    <footer>Pull quote caption</footer>
</blockquote>
<!-- /wp:core/pullquote -->
```

### PR DESCRIPTION
**Pull request by youknowriad**
closes #1982 

This PR seeks to solve the parsing/serializing issues we were experiencing when typing multiple paragraphs in pullquotes.

**Testing instructions**

 - Create a multi-p pullquote
 - Check the markup in the text editor (no nested paragraphs
 - Save and refresh, the markup should stay the same

### CODE REVIEW

**Review by afercia — DISMISSED** [review]
@youknowriad tested a bit and seems this approach perfectly works in Gutenberg.

However, when editing a post with a pullquote in the Classic editor, and then again in Gutenberg, the content gets lost. Not sure if this should go in a separate issue.

For example: inserted a pullquote with 2 paragraphs:
```
<blockquote class="wp-block-pullquote alignnone">
    <p>My quote</p>
    <p>with multiple par</p>
    <footer>my caption</footer>
</blockquote>

```

Saved, then opened the same post using the Classic editor, switched to text mode and the markup was changed to:
```
<!-- wp:core/pullquote -->
<blockquote class="wp-block-pullquote alignnone">My quote

with multiple par

<footer>my caption</footer></blockquote>
<!-- /wp:core/pullquote -->
```

This perfectly works in the Classic editor, but when editing again in Gutenberg, it doesn't of course and the content gets stripped out.

Marking as "Request changes" just because I'm not sure if you want to open a new issue.

**Comment by nylen** [comment]
@afercia that is part of a much larger class of issues with the way we parse block content.  Can you leave that comment on https://github.com/WordPress/gutenberg/issues/1736 as an example?

**Comment by nylen** [comment]
0c3e5ac fixes the previously existing tests.

2f96b6c adds a new test for this condition.  Strangely, this was being parsed and serialized correctly during the test suite - whatever issue caused the duplicate `p` tags was only appearing in actual usage, which is a bit concerning as it means there's some behavior we're not successfully testing.

Adding that new test surfaced a warning which I'm able to reproduce by using the editor:

- Create a multi-paragraph pullquote block.
- Switch from Visual to Text mode.  During serialization, the following warning is generated:

```
Warning: Each child in an array or iterator should have a unique "key" prop.

Check the top-level render call using <blockquote>. See https://fb.me/react-warning-keys for more information.
    in p
```

**Comment by youknowriad** [comment]
I fixed the warning, I'm not totally satisfied with the fix but it has to do with the `Editable` shape lacking `keys`. The `Editable` shape is part of a much larger concern.

**Comment by nylen** [comment]
There is an untested branch in the latest commit. How about fixing that by adding a `<strong>` or something in one of the paragraphs for the multi-paragraph test fixture?

I would like to know why this bug was occurring in the UI but not the test suite. Any thoughts there?

**Comment by youknowriad** [comment]
> I would like to know why this bug was occurring in the UI but not the test suite. Any thoughts there?

I thought about this and here is my hypothesis:
 - The parsing was dropping the `p` tag and leaving only the children
 - The serializer (save) was adding the `p` again

But when we use TinyMCE (the editable), for each item of the array, it adds a `p` wrapper if no wrapper is found (simple string) and when serialized, we end up with the double `p`.

**Comment by youknowriad** [comment]
Added a `strong` to the multi-paragraph example. Is this ready to be merged?

**Review by nylen — APPROVED** [review]
🚢

File: blocks/test/fixtures/core__pullquote__multi-paragraph.html [hunk 1]
```diff
@@ -0,0 +1,7 @@
+<!-- wp:core/pullquote -->
+<blockquote class="wp-block-pullquote alignnone">
+    <p>Paragraph <strong>one</strong></p>
+    <p>Paragraph two</p>
+    <footer>by whoever</footer>
```
Comment (westonruter) [inline blocks/test/fixtures/core__pullquote__multi-paragraph.html]: _whomever_ 🤓

**Pull request merged**